src/lib/bits/calendar/components/calendar.svelte

```diff
@@ -78,9 +80,15 @@
 		},
 		onValueChange: ({ next }: { next: $$Props["value"] }) => {
 			if (Array.isArray(next)) {
-				if (JSON.stringify(next) !== JSON.stringify(value)) {
+				if (!Array.isArray(value)) {
+					onValueChange?.(next);
+					value = next;
+					return next;
+				}
+				if (!arraysAreEqual(value, next)) {
 					onValueChange?.(next);
 					value = next;
+					return next;
```

Comment:
This could probably just be simplified to this:
```suggestion
				if (!Array.isArray(value) || !arraysAreEqual(value, next)) {
					onValueChange?.(next);
					value = next;
					return next;
```